src/lib/content/coordinators/FilterBy.ts

```diff
@@ -0,0 +1,216 @@
+import { AxiosInstance } from 'axios';
+import { createContentClient } from '../../client/createContentClient';
+import { ContentClientConfigV2 } from '../../config/ContentClientConfigV2';
+
+export interface IPageResponse<Body> {
+  responseCount: number;
+  nextCursor?: string;
+  next?: () => Promise<FilterByResponse<Body>>;
+}
+
+export interface FilterByResponse<Body> {
+  responses: ContentItemResponse<Body>[];
+  page: IPageResponse<Body>;
+}
+
+export interface ContentItemResponse<Body = any> {
+  content: Body & {
+    _meta: {
+      name: string;
+      schema: string;
+      deliveryId: string;
+      deliveryKey?: string;
+      parentId?: string;
+    };
+  };
+}
+
+export interface IFilterBy {
+  path: string;
+  value: any;
+}
+
+export type IOrder = 'DESC' | 'ASC';
+
+export interface ISortBy {
+  key: string;
+  order: IOrder;
+}
+
+export interface RequestOptions {
+  depth: 'all' | 'root';
```

Comment:
All should be optional. I might want to set depth, but keep format standard, for e.g.